core/src/consensus/stake/mod.rs

```diff
@@ -384,8 +386,12 @@ pub fn ban(state: &mut TopLevelState, criminal: Address) -> StateResult<()> {
     let mut jailed = Jail::load_from_state(state)?;
     let mut validators = Validators::load_from_state(state)?;
 
-    candidates.remove(&criminal);
-    jailed.remove(&criminal);
+    let deposit = match (candidates.remove(&criminal), jailed.remove(&criminal)) {
+        (Some(_), Some(_)) => unreachable!(),
```

Comment:
There should be a reason why this is unreachable.

Reply:
fixed